AsyncDisplayKit/ASTextNode.mm

```diff
@@ -76,7 +71,7 @@ @implementation ASTextNode {
   NSRange _highlightRange;
   ASHighlightOverlayLayer *_activeHighlightLayer;
 
-  ASDN::Mutex _rendererLock;
+  std::recursive_mutex _textLock;
```

Comment:
Got it, sounds good. So far as I could tell your locking with the recursive mutex is correct in this diff, but of course... there be dragons in multithreaded locking with recursive mutexes, so there may be edge cases. I guess I'm a little worried about rushing the threading model of one of the core classes of ASDK, but I guess if this is the interim fix I'm cool with that.

We probably need another few thorough reviews on this to verify the locking order is correct and we don't introduce hard-to-catch deadlocks.